src/meta.rs

```diff
@@ -174,10 +167,182 @@ impl Meta {
                         _ => (),
                     }
                 }
-            })
-            .map(|_| ());
+                drop(cfg_write);
+            });
         task
     }
+
+    pub fn monitor_metabucket(&self) {
+        let read_cfg = self.config.read().unwrap();
+
+        let metadata_bucket = read_cfg.server.metadata_bucket.clone();
+        let metadata_endpoint = read_cfg.server.metadata_endpoint.clone();
+        let access_key = read_cfg.server.access_key.clone();
+        let secret_key = read_cfg.server.secret_key.clone();
+        drop(read_cfg);
+
+        let mut c = minio::Client::new(&metadata_endpoint).expect("Could not connect metabucket");
+        c.set_credentials(Credentials::new(&access_key, &secret_key));
+
+        let cfg = Arc::clone(&self.config);
+        let task = c
+            .listen_bucket_notificaion(
+                &metadata_bucket,
+                None,
+                None,
+                vec![
+                    "s3:ObjectCreated:*".to_string(),
+                    "s3:ObjectRemoved:*".to_string(),
+                ],
+            )
+            .map_err(|_| ())
```

Comment:
Don't we need to implement a retry loop in case of connection errors?

Reply:
SDK should implement that, the retry is something i didn't implement from the `minio-go` sdk

Reply:
Cool, approving.